### PR TITLE
feat: implement `bazel-module` VersioningApi

### PR DESCRIPTION
**Pull request by cgrindel**
## Changes

Add VersioningApi for `bazel-module`.

## Context

Related to #13658.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by cgrindel** [comment]
Closes cgrindel/renovate_bzlmod_support#13.

File: lib/modules/versioning/bazel-module/index.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
+import type { SemVer } from 'semver';
+import type { NewValueConfig, VersioningApi } from '../types';
+import { BzlmodVersion } from './bzlmod-version';
+
+export const id = 'bazel-module';
+export const displayName = 'Bazel Module';
+export const urls = ['https://bazel.build/external/module'];
+export const supportsRanges = false;
+
+function getBzlmodVersion(version: string | SemVer): BzlmodVersion {
```
Comment (viceice) [inline lib/modules/versioning/bazel-module/index.ts:10]:
```suggestion
function getBzlmodVersion(version: string): BzlmodVersion {
```
remove, we'll never pass a SemVer object

File: lib/modules/versioning/bazel-module/bzlmod-version.spec.ts [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+      expect(ident.asNumber).toBe(0);
+      expect(ident.isDigitsOnly).toBe(false);
+    });
+
+    it.each([
+      { a: '1', b: '1', exp: true },
+      { a: '1', b: '2', exp: false },
+      { a: 'foo1', b: '1', exp: false },
+      { a: 'a', b: 'a', exp: true },
+      { a: 'a', b: 'b', exp: false },
```
Comment (zharinov) [inline lib/modules/versioning/bazel-module/bzlmod-version.spec.ts:24]: Please, use template literal syntax. Search our codebase for:
```
each`
```

Comment (cgrindel) [reply lib/modules/versioning/bazel-module/bzlmod-version.spec.ts:24]: Sorry. I missed that bullet point in the best-practices.

**Review by zharinov — DISMISSED** [review]
Thank you, I like the code quality 👍

**Review by rarkins — DISMISSED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**